test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Object.Tests.ps1

```diff
@@ -242,6 +243,30 @@ Describe "Select-Object DRT basic functionality" -Tags "CI" {
 		$results.Count | Should -Be 1
 		$results[0] | Should -BeExactly "3"
 	}
+
+    It "Select-Object should handle dynamic (DLR) properties"{
+        $dynObj = [TestDynamic]::new()
+        $results = $dynObj, $dynObj | Select-Object -ExpandProperty FooProp
+        $results.Count | Should -Be 2
+        $results[0] | Should -Be 123
+        $results[1] | Should -Be 123
+    }
+
+    It "Select-Object should handle dynamic (DLR) properties without GetDynamicMemberNames hint"{
+        $dynObj = [TestDynamic]::new()
+        $results = $dynObj, $dynObj | Select-Object -ExpandProperty HiddenProp
+        $results.Count | Should -Be 2
+        $results[0] | Should -Be 789
+        $results[1] | Should -Be 789
+    }
+
+    It "Select-Object should handle wildcarded dynamic (DLR) properties when hinted by GetDynamicMemberNames"{
+        $dynObj = [TestDynamic]::new()
+        $results = $dynObj, $dynObj | Select-Object -ExpandProperty Foo*
```

Comment:
Maybe also test `*Prop` (should be error with `-ExpandProperty`, maybe also test without `-ExpandProperty`.)

Reply:
Good ideas; will do.